source/Editor.js

```diff
@@ -1993,11 +1993,49 @@ var keyHandlers = {
             // If the commonAncestor was in a mathjax equation, delete the parent. Namely the mathjax equation.
             if (range.commonAncestorContainer && 
                 range.commonAncestorContainer.parentElement && 
-                checkInSVG.test(range.commonAncestorContainer.nodeName)
+                (checkInSVG.test(range.commonAncestorContainer.nodeName) || checkInSVG.test(range.commonAncestorContainer.parentElement.nodeName))
```

Comment:
It'd be good if you can make the condition of this if statement clearly express that; i.e. do something like the while loop below (terminating at the top of the body or something) to see if the commonAncestorContainer is a child of span.mathjax